modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/renewals-chart/index.js

```diff
@@ -55,6 +57,27 @@ export default function () {
 		getRenewalsData();
 	}, []);
 
+	const setNewArray = () => {
+		data?.items?.map((item) => {
+			const expirationInTime = new Date(item.closeDate) - currentDate;
+
+			const expirationInDays =
+				Math.floor(expirationInTime / milisecondsPerDay) + 1;
+
+			newArray.push({
+				closeDate: item.closeDate,
+				expirationDays: expirationInDays,
+				opportunityName: item.opportunityName,
+			});
+		});
+	};
+
+	setNewArray();
+
+	const filteredArray = newArray
+		.filter((item) => item.expirationDays > 0 && item.expirationDays <= 30)
+		.slice(0, 4);
```

Comment:
```suggestion
	const filteredArray = newArray
		.filter((item) => item.expirationDays > 0 && item.expirationDays <= 30)
		.slice(0, 4);
```

Importante colocar loops em algum hook como useMemo ou useCallback